### DISCUSSION TITLE
Why height of tree is one?

### DISCUSSION THREAD

**Discussion by Piyushbalwani** [online-ml/river#645] [Q&A]
Hi, 

If height of `HoeffdingTreeClassifier` is 1, what does it implies? Is data not sufficient to converge or data is not good? 
This problem is reproducible for many small datasets. The accuracy and F1-score is good for the same `HoeffdingTreeClassifier` model, but still `None` dataframe returned by the model.
Data is sufficient for the sklearn `DecisionTreeClassifier` and it seems to be fit nicely and have decent height for prediction. 

**Note**: accuracy and f1 score of `HoeffdingTreeClassifier` is calculated during sample wise learning of the classifier.

Thanks!!

**Comment by jacobmontiel** [comment]
Without information about the data (size, type, etc.) it is hard to provide a precise explanation. In the example that you describe, it is likely that the tree has not "seen" enough information to make a split. The batch Decision Tree classifier, on the other hand, may grow as it has access to all the data. The Hoeffding Tree model approaches the performance of the batch Decision Tree model as the number of samples increases.

In other words, the Hoefdding Tree grows as it gets more data, and the split criterium is met (Hoeffding bound).
